woocommerce/woocommerce
# Add customize store assembler hub

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #39714.

This PR implements the Assembler Hub main screen of the CYS project.

https://github.com/woocommerce/woocommerce/assets/4344253/efbca9c5-734b-4882-88fc-d89d77eb0a08

- Figma: Mrk6SERPZ4KrFHSjM0a8TK-fi-2631%3A253794

- I turned off many ts because our wp packages don't support typescript. :( 
- The implementation is based on Gutenberg's site editor. I forked the components and modified them to meet our requirements. I didn't refactor the component or make any large changes because when Gutenberg is changed, I think we could then easily compare it, and in the future, we will contribute back to Gutenberg.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use WP 6.3 and Gutenberg 16.4 (Currently only developed and tested based on this version, should also work without Gutenberg plugin but haven't fully tested it.)
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Go to `WooCommerce > Home`
5. Click on `Customize your store` task 
6. Click on `Assembler Hub` button
7. You should see the Assembler Hub screen
8. Confirm the UI matches the design (Mrk6SERPZ4KrFHSjM0a8TK-fi-2631%3A253794). Please note that we do not support mobile for now.
9. You should be able to navigate to the relevant subpage by clicking on the sidebar menu item
10. Frame should be resizable (Resizing frame is a bit laggy, I'll create a follow-up issue to optimize it) and browsable 
11. Click on "back" icon on top menu
12. Should be redirected to "Design with AI step"
13. Enable XState dev tool (`window.localStorage.setItem( 'xstate_inspect', 'true')`) and go back to `Assembler Hub
14. Click on "Done" button
15. Observe that `backToHomescreen` is the active state in XState Visualizer.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add customize store assembler hub

#### Comment 

</details>
